acquia_lift_profiles/acquia_lift_profiles.admin.inc

```diff
@@ -101,43 +101,37 @@ function acquia_lift_profiles_admin_form($form, &$form_state) {
     )),
     '#default_value' => variable_get('acquia_lift_profiles_default_identity_type', ''),
   );
-  $vocabularies = taxonomy_get_vocabularies();
-  $mappings = variable_get('acquia_lift_profiles_vocabulary_mappings', array());
-  $vocab_options = array('' => t('Select...'));
-  foreach ($vocabularies as $vid => $voc) {
-    $vocab_options[$vid] = $voc->name;
-  }
+
+  // Retrieving all context options.
+  module_load_include('inc', 'personalize', 'personalize.admin');
+  $groups = personalize_get_grouped_context_options(NULL, TRUE, array('acquia_lift_profiles_context', 'acquia_lift_context'));
+
+  // Vocabulary mappings.
   $form['acquia_lift_profiles_vocabulary_mappings'] = array(
     '#title' => t('Vocabulary Mappings'),
+    '#description' => t('For each vocabularies available in Acquia Lift Profiles, you can map a visitor context, whose value will then be sent as the value for that vocabulary.'),
     '#type' => 'fieldset',
     '#tree' => TRUE,
   );
-  $form['acquia_lift_profiles_vocabulary_mappings']['content_section'] = array(
-    '#title' => t('Content Section'),
-    '#description' => t('The vocabulary to use for the content section'),
-    '#type' => 'select',
-    '#options' => $vocab_options,
-    '#default_value' => isset($mappings['content_section']) ? $mappings['content_section'] : '',
-  );
-  $form['acquia_lift_profiles_vocabulary_mappings']['content_keywords'] = array(
-    '#title' => t('Content Keywords'),
-    '#description' => t('The vocabulary to use for the content keywords'),
-    '#type' => 'select',
-    '#options' => $vocab_options,
-    '#default_value' => isset($mappings['content_keywords']) ? $mappings['content_keywords'] : '',
-  );
-  $form['acquia_lift_profiles_vocabulary_mappings']['persona'] = array(
-    '#title' => t('Persona'),
-    '#description' => t('The vocabulary to use for the persona'),
-    '#type' => 'select',
-    '#options' => $vocab_options,
-    '#default_value' => isset($mappings['persona']) ? $mappings['persona'] : '',
+  $type_list = array(
+    'content_section',
+    'content_keywords',
+    'persona',
   );
+  $vocabulary_mappings = variable_get('acquia_lift_profiles_vocabulary_mappings', array());
+  foreach ($type_list as $type) {
+    $default_value = isset($vocabulary_mappings[$type]) ? $vocabulary_mappings[$type] : NULL;
+    $select = acquia_lift_profiles_admin_build_visitor_context_select($groups, $default_value);
+    $type_name = str_replace('_', ' ', $type);
+    $type_title = ucfirst($type_name);
```

Comment:
Let's make $type_list an associative array with machine names as keys and friendly names as values.